diff --git a/node_modules/@docusaurus/theme-search-algolia/src/theme/SearchBar/index.js b/node_modules/@docusaurus/theme-search-algolia/src/theme/SearchBar/index.js
index 0103602..2046cc0 100644
--- a/node_modules/@docusaurus/theme-search-algolia/src/theme/SearchBar/index.js
+++ b/node_modules/@docusaurus/theme-search-algolia/src/theme/SearchBar/index.js
@@ -5,16 +5,16 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import React, {useState, useRef, useCallback, useMemo} from 'react';
-import {createPortal} from 'react-dom';
-import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
-import {useHistory} from '@docusaurus/router';
-import {useBaseUrlUtils} from '@docusaurus/useBaseUrl';
-import Link from '@docusaurus/Link';
+import { DocSearchButton, useDocSearchKeyboardEvents } from '@docsearch/react';
 import Head from '@docusaurus/Head';
+import Link from '@docusaurus/Link';
+import { useHistory } from '@docusaurus/router';
+import { useBaseUrlUtils } from '@docusaurus/useBaseUrl';
+import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
 import useSearchQuery from '@theme/hooks/useSearchQuery';
-import {DocSearchButton, useDocSearchKeyboardEvents} from '@docsearch/react';
-import {useAlgoliaContextualSearchParameters} from '../../utils/algoliaSearchUtils';
+import React, { useCallback, useMemo, useRef, useState } from 'react';
+import { createPortal } from 'react-dom';
+import { useAlgoliaContextualSearchParameters } from '../../utils/algoliaSearchUtils';
 
 let DocSearchModal = null;
 
@@ -85,7 +85,7 @@ function DocSearch({contextualSearch, ...props}) {
 
   const navigator = useRef({
     navigate({suggestionUrl}) {
-      history.push(suggestionUrl);
+      window.location.assign(suggestionUrl);
     },
   }).current;
 
